test/tools/fuzzTesting/createRandomTest.cpp

```diff
@@ -35,364 +35,364 @@ namespace dev { namespace test {
-			else if (type == "[0xHASH32]")
-				replace = "0x" + rndByteSequence(32);
-			else if (type == "[V]")
-			{
-				int random = randomPercent();
-				if (random < 30)
-					replace = "0x1c";
-				else if (random < 60)
-					replace = "0x1d";
-				else
-					replace = "0x" + rndByteSequence(1);
-			}
-			else if (type == "[BLOCKGASLIMIT]")
-				replace = randomUniIntHex(dev::u256("100000"), dev::u256("36028797018963967"));
-			else if (type == "[DESTADDRESS]")
-			{
-				Address address = _options.getRandomAddress(RandomCodeOptions::AddressType::PrecompiledOrStateOrCreate);
-				if (address != ZeroAddress)
-					replace = "0x" + toString(address);
-				else
-					replace = std::string(); // transaction creation
-			}
-			else if (type == "[ADDRESS]")
-			{
-				Address destAddress = _options.getRandomAddress(RandomCodeOptions::AddressType::PrecompiledOrState);
-				replace = toString(destAddress);
-			}
-			else if (type == "[0xADDRESS]") {
-				replace = "0x" + toString(_options.getRandomAddress(RandomCodeOptions::AddressType::StateAccount));
-			}
-			else if (type == "[TRANSACTIONGASLIMIT]")
-				replace = randomUniIntHex(dev::u256("25000"), dev::u256("10000000"));
-			else if (type == "[GASPRICE]")
-				replace = randomUniIntHex(0, dev::u256("10"));
-			else
-			{
-				//Replace type from varMap if varMap is set
-				if (_varMap.count(type))
-					replace = _varMap.at(type);
-				else
-					BOOST_ERROR("Skipping undeclared type: " + type);
-			}
+    for (auto const& type: types)
+    {
+        std::size_t pos = _test.find(type);
+        while (pos != std::string::npos)
+        {
+            std::string replace;
+            if (type == "[RLP]")
+            {
+                std::string debug;
+                int randomDepth = 1 + (int)randomSmallUniInt() % 10;
+                replace = rndRLPSequence(randomDepth, debug);
+                cnote << debug;
+            }
+            else if (type == "[CODE]")
+                replace = generate(50, _options);
+            else if (type == "[HEX]")
+                replace = randomUniIntHex();
+            else if (type == "[HEX32]")
+                replace = randomUniIntHex(0, std::numeric_limits<uint32_t>::max());
+            else if (type == "[HASH20]")
+                replace = rndByteSequence(20);
+            else if (type == "[HASH32]")
+                replace = rndByteSequence(32);
+            else if (type == "[0xHASH32]")
+                replace = "0x" + rndByteSequence(32);
+            else if (type == "[V]")
+            {
+                int random = randomPercent();
+                if (random < 30)
+                    replace = "0x1c";
+                else if (random < 60)
+                    replace = "0x1d";
+                else
+                    replace = "0x" + rndByteSequence(1);
+            }
+            else if (type == "[BLOCKGASLIMIT]")
+                replace = randomUniIntHex(dev::u256("51000000"), dev::u256("36028797018963967"));
```

Comment:
This value does not make sense.

Reply:
min or max?

Reply:
The min (the changed one). It's quite far from real values.

Why is this change needed?

Reply:
so that transaction wont go OOG intrinsic

` maxTransactionGasPrice * maxTransactionGasLimit  < minBlockGasLimit `

Reply:
So what are the values of maxTransactionGasPrice and maxTransactionGasLimit?

Reply:
maxTransactionGasPrice = 5
maxTransactionGasLimit = 10 000 000   (Vitalik's theoretical block gas limit for the near future)

Reply:
Ok, so I'd change it to 50000001.
Secondly, you don't need string literals to create these values.